diff --git a/include/jemalloc/internal/BUILD.bazel b/include/jemalloc/internal/BUILD.bazel
--- a/include/jemalloc/internal/BUILD.bazel
+++ b/include/jemalloc/internal/BUILD.bazel
@@ -223,6 +223,31 @@
     ],
 )
 
+selects.config_setting_group(
+    name = "jemalloc_lg_sizeof_long_32_bit",
+    match_any = [
+        "@platforms//os:windows",
+        "@platforms//cpu:aarch32",
+        "@platforms//cpu:x86_32",
+    ],
+)
+
+selects.config_setting_group(
+    name = "jemalloc_lg_sizeof_long_64_bit_cpu",
+    match_any = [
+        "@platforms//cpu:aarch64",
+        "@platforms//cpu:x86_64",
+    ],
+)
+
+selects.config_setting_group(
+    name = "jemalloc_lg_sizeof_long_64_bit",
+    match_all = [
+        "//settings/platform:posix",
+        ":jemalloc_lg_sizeof_long_64_bit_cpu",
+    ],
+)
+
 configure_header(
     name = "jemalloc_internal_defs",
     out = "jemalloc_internal_defs.h",
@@ -456,11 +481,8 @@
         define_macro_with("LG_SIZEOF_LONG_LONG", "3") |
         select(
             {
-                "@platforms//os:windows": define_macro_with("LG_SIZEOF_LONG", "2"),
-                "@platforms//cpu:aarch32": define_macro_with("LG_SIZEOF_LONG", "2"),
-                "@platforms//cpu:aarch64": define_macro_with("LG_SIZEOF_LONG", "3"),
-                "@platforms//cpu:x86_32": define_macro_with("LG_SIZEOF_LONG", "2"),
-                "@platforms//cpu:x86_64": define_macro_with("LG_SIZEOF_LONG", "3"),
+                ":jemalloc_lg_sizeof_long_32_bit": define_macro_with("LG_SIZEOF_LONG", "2"),
+                ":jemalloc_lg_sizeof_long_64_bit": define_macro_with("LG_SIZEOF_LONG", "3"),
             },
             no_match_error = "Determined by ABI, support for less common target cpus not yet implemented",
         ) |
