diff --git a/include/jemalloc/jemalloc_macros.h.in b/include/jemalloc/jemalloc_macros.h.in
--- a/include/jemalloc/jemalloc_macros.h.in
+++ b/include/jemalloc/jemalloc_macros.h.in
@@ -51,7 +51,8 @@
  */
 #define MALLCTL_ARENAS_DESTROYED	4097
 
-#if defined(__cplusplus) && defined(JEMALLOC_USE_CXX_THROW)
+#if defined(__cplusplus) && defined(JEMALLOC_USE_CXX_THROW) && \
+    (defined(__GLIBC__) || defined(__FreeBSD__))
 #  define JEMALLOC_CXX_THROW throw()
 #else
 #  define JEMALLOC_CXX_THROW
@@ -145,5 +146,9 @@
 #if (defined(__APPLE__) || defined(__FreeBSD__)) && !defined(JEMALLOC_NO_RENAME)
 #  define JEMALLOC_SYS_NOTHROW
 #else
-#  define JEMALLOC_SYS_NOTHROW JEMALLOC_NOTHROW
+#  if defined(__cplusplus) && !defined(JEMALLOC_NO_RENAME) && !defined(__GLIBC__)
+#    define JEMALLOC_SYS_NOTHROW
+#  else
+#    define JEMALLOC_SYS_NOTHROW JEMALLOC_NOTHROW
+#  endif
 #endif
